src/lib/components/NglMoleculeViewer.react.js

```diff
@@ -0,0 +1,274 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Stage, Selection} from 'ngl';
+
+/**
+ * The NglMoleculeViewer is used to render schematic diagrams
+ * of biomolecules in ribbon-structure representations.
+ * Read more about the component here:
+ * https://github.com/IvoLeist/dash_ngl
+ * Read more about the used WebGL protein viewer here:
+ * https://github.com/arose/ngl
+ */
+export default class NglMoleculeViewer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {stage: null, structuresList: []};
+    }
+
+    componentDidMount() {
+        const {id, stageParameters, viewportStyle} = this.props;
+        const params = {...stageParameters};
+        const stage = new Stage(id, params);
+        stage.setSize(viewportStyle.width, viewportStyle.height);
+        this.setState({stage});
+    }
+
+    shouldComponentUpdate(prevProps, nextProps) {
+        const {stageParameters, data} = this.props;
+
+        // check if data has changed
+        if ((data !== null) & (prevProps.data !== null)) {
```

Comment:
```suggestion
        if ((data !== null) && (prevProps.data !== null)) {
```
A single ampersand is a bitwise operation for two numbers (https://stackoverflow.com/questions/7310109/whats-the-difference-between-and-in-javascript). I think you meant to use `&&` which is the usual Boolean operator.

Reply:
good catch thank you :)